### PR TITLE
Remove tls sni in nginx plugin

### PR DESCRIPTION
**Pull request by adferrand**
This PR is a part of the tls-sni-01 removal plan described in #6849.

It removes the capability to make tls-sni-01 challenges in the nginx plugin.

One subtility to note here. Nginx plugin was consuming the `--tls-sni-01-port` cli flag value (default 443) to handle tls-sni challenges of course, but also as the https port to use when relevant vhosts are setup/created with the issue certificate deployed on them. Most of the time, there is not reason that https port and tls-sni port should be different and not equals to 443. But `--tls-sni-01-port` was used (even if not accurately said by the documentation) to have a specific https port for nginx unit and integration tests.

Consequently, as this flag will be removed with #6853, I add a new flag focused on the specific purpose to define what is the https port for the Nginx plugin, `--https-port`.

### CODE REVIEW

**Comment by adferrand** [comment]
Blocking on #6854

**Comment by adferrand** [comment]
Counter-proposal.

The flag `--tls-sni-01` cannot be set (if used) to anything except `443` on production. Because this is the port used by ACME CA servers on production. It is useful only in the CI processes, when the local ACME CA is set to use a non standard port for that challenge, like with Boulder and Pebble.

Also, I think that the fact `--tls-sni-01` influencing the port used to serve real HTTPS connections for Nginx configured through the nginx plugin, is an undocumented and unexpected feature due to a flaw in the design of this plugin.

In fact, the Apache plugin is not designed like this, and always use the `443` to serve HTTPS connections. The value of `--tls-sni-01` is used only for vhost set to fulfill tls-sni challenges.

So I propose to not put a new flag `--https-port`, then hardcode `443` for SSL vhosts in nginx plugin like in apache plugin, and let the `--tls-sni-01` become a noop flag here for nginx, and more generally for certbot with #6853.

**Comment by bmw** [comment]
It's definitely a niche thing, but people may be using alternate ports for custom uses where they want SSL on the traffic perhaps with something forwarding port 443 traffic to this alternate port. Here's [an issue](https://github.com/certbot/certbot/issues/1754) asking for us to support this in the Apache plugin and some of the people in https://github.com/certbot/certbot/issues/5517 have the same problem.

While I think it's quite possible that no one is using this functionality, I don't think we have a way to know that that's the case. Looking through git history, the nginx plugins seems to have been doing things this way for a long time. Because of this, I don't feel good about suddenly ripping it out.

We could potentially deprecate the feature and then remove it after a few releases and see if the demand pops up, but since people have wanted similar functionality in the Apache plugin and it already exists in nginx, my personal opinion is it's worth keeping around and eventually adding to Apache.

**Comment by adferrand** [comment]
Yes it seems a feature that could be useful. To you want me to add an issue to do that for Apache in the wishlist?

**Comment by bmw** [comment]
I think that's covered by https://github.com/certbot/certbot/issues/1754.

File: certbot-nginx/certbot_nginx/configurator.py [hunk 1]
```diff
@@ -602,16 +600,18 @@ def _get_snakeoil_paths(self):
...
-    def _make_server_ssl(self, vhost):
+    def _make_server_ssl(self, vhost, https_port):
         """Make a server SSL.
 
         Make a server SSL by adding new listen and SSL directives.
 
         :param vhost: The vhost to add SSL to.
         :type vhost: :class:`~certbot_nginx.obj.VirtualHost`
+        :param https_port: The HTTPS port to use
+        :type vhost: int
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-nginx/certbot_nginx/configurator.py]: Not relevant anymore.

File: certbot-nginx/certbot_nginx/configurator.py [hunk 2]
```diff
@@ -602,16 +600,18 @@ def _get_snakeoil_paths(self):
             cert_file.write(cert_pem)
         return cert_path, le_key.file
 
-    def _make_server_ssl(self, vhost):
+    def _make_server_ssl(self, vhost, https_port):
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-nginx/certbot_nginx/configurator.py]: Indeed. Changed to use internal self.config

File: certbot-nginx/certbot_nginx/configurator.py [hunk 3]
```diff
@@ -933,9 +931,6 @@ def get_version(self):
         version_regex = re.compile(r"nginx version: ([^/]+)/([0-9\.]*)", re.IGNORECASE)
         version_matches = version_regex.findall(text)
 
-        sni_regex = re.compile(r"TLS SNI support enabled", re.IGNORECASE)
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-nginx/certbot_nginx/configurator.py]: Fixed. With the relevant unit tests back.

File: certbot-nginx/certbot_nginx/configurator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-nginx/certbot_nginx/configurator.py]: Not relevant anymore

**Pull request merged**